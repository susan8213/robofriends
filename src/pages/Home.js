import React, { useEffect } from 'react'
import ReactLoading from "react-loading";
import CardGallery from "../components/CardGallery"
import SearchBar from "../components/SearchBar"
import Scroll from "../components/Scroll"
import "tachyons";


function Home({ onRequestRobot, onSearchChange, robots, searchField, isPending, error }) {
    const filteredRobots = robots.filter(robot => robot.name.toLowerCase().includes(searchField.toLowerCase()));
  
    useEffect(() => {
      onRequestRobot();
    }, [onRequestRobot]);
  
    useEffect(() => {
      if (error) alert(error);
    }, [error]);
  
  
    return (
      <div className="App tc">
        <h1 className="f1 App-header">RoboFriends</h1>
        <SearchBar handleSearchChange={onSearchChange} />
        {
          isPending ? (
            <ReactLoading className="center" type={"bubbles"} color={"#cdecff"} height={'10%'} width={'10%'} />
          ) : (
            <Scroll>
              <CardGallery robots={filteredRobots} />
            </Scroll>
          )
        }
      </div>
    );
}

export default Home;