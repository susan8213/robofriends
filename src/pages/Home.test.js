import { shallow, render } from 'enzyme';
import React from 'react';
import Home from './Home';

let wrapper;

describe('<Home />', () =>{
    beforeEach(() => {
        const mockProps = {
            onRequestRobots: jest.fn(),
            robots: [],
            searchField: '',
            isPending: false
          }
          wrapper = shallow(<Home {...mockProps}/>)
    })

    it('render without crashing', () => {
        expect(wrapper.html()).toMatchSnapshot();
    })

    it('filter robots correctly', () => {
        const mockProps = {
            onRequestRobots: jest.fn(),
            robots: [{id: 1, name: 'Leanne Graham'}],
            searchField: 'Leanne',
            isPending: false
          }
          wrapper = render(<Home {...mockProps}/>)
          expect(wrapper.find('.card')).toHaveLength(1);
    })

    it('filter robots correctly with 0 result', () => {
        const mockProps = {
            onRequestRobots: jest.fn(),
            robots: [{id: 1, name: 'Leanne Graham'}],
            searchField: 'abcde',
            isPending: false
          }
          wrapper = render(<Home {...mockProps}/>)
          expect(wrapper.find('.card')).toHaveLength(0);
    })


})