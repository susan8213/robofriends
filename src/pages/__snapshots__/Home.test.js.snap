// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> render without crashing 1`] = `"<div class=\\"App tc\\"><h1 class=\\"f1 App-header\\">RoboFriends</h1><div class=\\"pa2\\"><input type=\\"search\\" class=\\"pa3 ba b--lightest-blue bg-white\\" placeholder=\\"Search robots\\"/></div><div style=\\"overflow:scroll;height:80vh\\"></div></div>"`;
